appGeo/src/app/game/game-rules.component.ts

```diff
@@ -0,0 +1,202 @@
+
+        this.maxSoloKills = MIN_MAX_KILL
+        this.maxGlobalKills = DEF_MAX_GLOBAL_KILL
+    }
+
+    isScheduledEnd() {
+        return this.scheduledEnd
+    }
+
+    isWipeoutEnd() {
+        return this.wipeOutEnd
+    }
+
+    setScheduledEnd(b: boolean) {
+        this.scheduledEnd = b
+    }
+
+    setWipeoutEnd(b: boolean) {
+        this.wipeOutEnd = b
+    }
+
+    //GETTERS
+    getMinPlayers() {
+        return this.minPlayers
+    }
+
+    getFractionKillers() {
+        return this.fractionKillers
+    }
+
+    getGameLengthHours() {
+        return (this.gameLength / 60)
+    }
+
+    getDayCycleLength() {
+        return this.dayCycleLength
+    }
+
+    getSafeLength() {
+        return this.safeLength
+    }
+
+    getVoteLength() {
+        return this.voteLength
+    }
+
+    getVoteTime() {
+        return this.voteTime
+    }
+
+    getMaxSoloKill() {
+        return this.maxSoloKills
+    }
+
+    getMaxGlobalKill() {
+        return this.maxGlobalKills
+    }
+
+    //SETTERS
+    setMinPlayers(minPlayers: number) {
+        if(minPlayers < MIN_MIN_PLAYERS) {
+            return FAILURE
+        }
+
+        this.minPlayers = minPlayers
+        return SUCCESS
+    }
+
+    setFractionKillers(frac: number) {
+        if((frac > 1) || (frac <= 0)) {
+            return FAILURE
+        }
+
+        this.fractionKillers = frac
+        return SUCCESS
+    }
+
+    setGameDurations(gameHours: number, cycleMinutes: number, 
+                     safeMinutes: number, voteMinutes: number) {
+        var gameMinutes = gameHours * 60
+
+        if(((gameMinutes) > MIN_GAME_LENGTH)) {
+            return FAILURE
+        } else if((cycleMinutes > gameMinutes / 6) || (cycleMinutes < 0)) {
+            return FAILURE
+        } else if((safeMinutes > cycleMinutes / 2) || (safeMinutes < 600)) {
+            return FAILURE
+        } else if((voteMinutes > safeMinutes) || (voteMinutes < 0)) {
+            return FAILURE
+        }
+
+        this.gameLength = gameMinutes
+        this.dayCycleLength = cycleMinutes
+        this.safeLength = safeMinutes
+        this.voteLength = voteMinutes
+        return SUCCESS
+    }
+
+    setVoteTime(minutesIntoCycle: number) {
+        if(minutesIntoCycle > (this.dayCycleLength - this.safeLength) || minutesIntoCycle < 0) {
+            return FAILURE
+        }
+
+        this.voteTime = minutesIntoCycle
+        return SUCCESS
+    }
+
+    setMaxSoloKill(maxKills: number) {
+        if(maxKills < MIN_MAX_KILL) {
+            return FAILURE
+        }
+        this.maxSoloKills = maxKills
+        return SUCCESS
+    }
+
+    setMaxGlobalKill(maxKills: number) {
+        if(maxKills < MIN_MAX_KILL) {
+            return FAILURE
+        }
+        this.maxGlobalKills = maxKills
+        return SUCCESS
+    }
+
+    isTestingOverrule() {
+        return this.testing_overrule
+    }
+
+    setTestingOverrule(b: boolean) {
+        this.testing_overrule = b
```

Comment:
Same as previous two comments about return values. Only mentioning this as it may come up for unit testing.